cranelift/codegen/src/isa/riscv64/inst/emit.rs

```diff
@@ -480,17 +411,35 @@ impl MachInstEmit for Inst {
             }
             &Inst::LoadConst32 { rd, imm } => {
                 let rd = allocs.next_writable(rd);
-                LoadConstant::U32(imm)
-                    .load_constant(rd, &mut |_| rd)
-                    .into_iter()
-                    .for_each(|inst| inst.emit(&[], sink, emit_info, state));
+                let label_data = sink.get_label();
+                let data = imm.to_le_bytes();
+
+                // Emit the constant into the pool
+                sink.defer_constant(label_data, 4, &data[..], u32::MAX);
```

Comment:
Why use a constant pool for 32bit values? Loading from the constant pool requires auipc+lw while loading a constant requires lui+addiw which is exactly the same size in terms of instructions but doesn't require an extra 4 bytes for the constant pool entry.

Reply:
We do try to use `lui`+`addi` in `Inst::load_constant_u32` and fall back here if those for some reason can't load it. Although I'm not entirely sure if that ever happens.

Reply:
That shouldn't be possible. 20 for lui + 12 for addiw == 32.

Reply:
That's right, and I think that just found a bug with our constant emissions, I tried to assert that we would always emit `lui`+`addi` and that assert failed! I'm going to investigate further, but yeah removing the LoadConst32 seems like the way to go.